core/drivers/tpm.c

```diff
@@ -0,0 +1,116 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2020, ARM Limited
+ */
+
+#include <kernel/generic_boot.h>
+#include <mm/core_mmu.h>
+#include <mm/core_memprot.h>
+#ifdef CFG_DT
+#include <kernel/dt.h>
+#include <libfdt.h>
+#endif
+#include <drivers/tpm.h>
+#include <string.h>
+
+#ifdef CFG_TPM_API
+
+#define PAGE_SIZE 4096
+
+#ifdef CFG_DT
+static const char *dt_tpm_match_table = {
+    "arm,nt_fw",
+};
+#endif
+
+static vaddr_t log_va_addr = 0;
+static size_t log_size = 0;
+
+void map_tpm_log_area(void *fdt)
+{
+    log_size = 0;
+    paddr_t log_phys_addr;
+
+#ifdef CFG_DT
```

Comment:
This should be moved into a helper function.

Reply:
This is actually something I thought about when I was implementing the functionality, but I wasn't sure where to put it. Should map_tpm_log_area() be moved to generic_boot.c as a helper function, for instance?